### ISSUE TITLE
[Feature Request] Make (heavy) vehicles destroy or deteriorate trenches

### ISSUE THREAD

**Issue by 0xlord** [gruppe-adler/grad_trenches#119]
Would be a great addition to the mod if vehicles would do (different) damage up to the point that heavy vehicles (think IFVs or MBTs) would destroy them. Would this be possible with the damaging system when shooting them?

**Comment by nomisum** [comment]
shooting trenches should be a thing already. check `allowHitDecay` CBA setting.

it only works if splash damage is > 1 and the exact formula is:
`((_hitValue + _indirectHitValue + _splashDamage)/3000);` multiplied by `hitDecayMultiplier` (another CBA setting)

**Comment by nomisum** [comment]
ah sorry, now i seem to understand - you want decay on "bumping" into the trench, right? should be possible, although physx contact EHs are a bit funky.

**Comment by 0xlord** [comment]
> shooting trenches should be a thing already. check allowHitDecay CBA setting
Yes, I know. Guessed the underlying system could be used for this though.

>you want decay on "bumping" into the trench, right
Yes, exactly. Sorry for my bad expression (:

> i would likely link the damage to the mass of the object + differentiate between tracked and wheeled.
Great idea for differentiating between vehicles. Would see an APC more going over rather than going through a trench. Should still be damaged. Maybe something like `mass + drive_system + speed`?

**Comment by nomisum** [comment]
Sounds good. I hope the engine will handle the animation change "at contact" + network synchronization somewhat gracefully, as Arma is notoriously good at creating tracked rockets. I cant promise any work on this in the near future though.

**Comment by 0xlord** [comment]
> I cant promise any work on this in the near future though.

No worries. Super happy that you even consider this. Looking forward to any news regarding this. Have nice Sunday evening :)
